File: Pages/login_page.js
/**
 * This class is responsible for the login
 * functionality testing.
 */

"use strict";
const { Builder, By, Key, until } = require('selenium-webdriver');
const config = require('../util/config.json');

class LoginPage {
    constructor(driver) {
        this.driver = driver;
    }

    async login() {
        await this.driver.get(config.site_to_test);
        await this.driver.findElement(By.name('username')).sendKeys(config.trading_admin_email, Key.RETURN);
        await this.driver.findElement(By.name('password')).sendKeys(config.password, Key.RETURN);
    }
}

module.exports = LoginPage;

File: Pages/loginPage.py
import time
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from config import LINKS


class LoginPage:
    URL = LINKS.SITE
    heading_xpath = (By.XPATH, f"//h2[@class='access__form__title']")
    heading_text = f"Login"
    username_xpath = (By.XPATH, f"//input[@data-test='username']")
    password_xpath = (By.XPATH, f"//input[@data-test='password']")
    submit_xpath = (By.XPATH, f"//button[@data-test='submit']")

    def __init__(self, browser):
        self.browser = browser

    def load(self):
        self.browser.get(self.URL)
        assert self.browser.find_element(*self.heading_xpath).text == self.heading_text

    def login(self, username, password):
        self.browser.find_element(*self.username_xpath).send_keys(username)
        self.browser.find_element(*self.password_xpath).send_keys(password)
        self.browser.find_element(*self.submit_xpath).click()
        time.sleep(2)
        assert ("dashboard" in self.browser.current_url) == True
